srcs/parsing/check_map_grid.c

```diff
@@ -5,125 +5,79 @@
 /*                                                    +:+ +:+         +:+     */
 /*   By: aweaver <aweaver@student.42.fr>            +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
-/*   Created: 2022/09/28 11:11:14 by aweaver           #+#    #+#             */
-/*   Updated: 2022/09/29 11:44:55 by aweaver          ###   ########.fr       */
+/*   Created: 2022/10/03 09:27:26 by aweaver           #+#    #+#             */
+/*   Updated: 2022/10/03 13:55:04 by aweaver          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "cub3d.h"
 
-t_list	*ft_skip_header(t_cub_data *data)
+static int	ft_is_player_starting_position(char c)
 {
-	t_map_data	*map;
-	t_list		*list;
-	int			i;
-
-	list = data->lst_map;
-	while (list)
-	{
-		map = (t_map_data *)list->content;
-		i = 0;
-		while (map->line && map->line[i])
-		{
-			if (ft_iswhitespace(map->line[i]))
-				i++;
-			else if (ft_isalpha(map->line[i]))
-				break ;
-			else if ((map->line[i]) == '0' || map->line[i] == '1')
-				return (list);
-		}
-		list = list->next;
-	}
-	return (list);
-}
-
-int	ft_create_str(char **tab, int tab_size, int length)
-{
-	int	i;
-
-	i = 0;
-	while (i < tab_size)
-	{
-		tab[i] = malloc(sizeof(*tab[i]) * (length + 1));
-		if (tab[i] == NULL)
-		{
-			while (--i >= 0)
-			{
-				free(tab[i]);
-				return (1);
-			}
-		}
-		i++;
-	}
-	tab[i] = NULL;
+	if (c == 'N' || c == 'E' || c == 'W' || c == 'S')
+		return (1);
 	return (0);
 }
 
-void	ft_fill_grid(t_list *list, int length)
+static void	ft_is_valid_map_pattern(t_cub_data *data, char **grid, int x, int y)
 {
-	t_map_data	*map;
-	t_cub_data	*data;
-	int			i;
-	int			j;
+	static int	player_position = 0;
 
-	map = (t_map_data *)list->content;
-	data = map->data;
-	i = 0;
-	while (list)
+	if (grid[y][x] == '1' || grid[y][x] == ' ' || grid[y][x] == '\n')
+		return ;
+	else
 	{
-		map = (t_map_data *)list->content;
-		j = -1;
-		while (map->line[++j])
-			data->grid[i][j] = map->line[j];
-		while (j < length)
-		{
-			data->grid[i][j] = '\0';
-			j++;
-		}
-		data->grid[i][j] = '\0';
-		i++;
-		list = list->next;
+		if (ft_is_player_starting_position(grid[y][x]))
+			player_position++;
+		if (player_position > 1)
+			ft_exit_check_grid(data, "Redefining player spawn position.\n");
```

Comment:
Could you please reformat this line of code ?

Reply:
I added the check for players straight from ft_check_map_grid.
Good thinking ;-)